spec/unit/fibered_database_connection_pool_spec.rb

```diff
@@ -392,38 +394,24 @@ def cancel_timer(timer_block)
 
     context "with more than 1 connection in the pool" do
       it "should serve separate connections per fiber" do
-        expected_query = if Rails::VERSION::MAJOR > 4
-                           "SET  @@SESSION.sql_mode = CONCAT(CONCAT(@@sql_mode, ',STRICT_ALL_TABLES'), ',NO_AUTO_VALUE_ON_ZERO'),  @@SESSION.sql_auto_is_null = 0, @@SESSION.wait_timeout = 2147483"
-                         else
-                           "SET  @@SESSION.sql_auto_is_null = 0, @@SESSION.wait_timeout = 2147483, @@SESSION.sql_mode = 'STRICT_ALL_TABLES'"
-                         end
-        expect(client).to receive(:query) do |*args|
-          expect(args).to eq([expected_query])
-        end.exactly(2).times
-
         c0 = ActiveRecord::Base.connection
         c1 = nil
-        fiber = Fiber.new { c1 = ActiveRecord::Base.connection }
+        fiber = Fiber.new do
+          c1 = ActiveRecord::Base.connection
+        end
         fiber.resume
 
-        expect(c0).to be
-        expect(c1).to be
+        expect(c0).to be_truthy
+        expect(c1).to be_truthy
         expect(c1).to_not eq(c0)
         expect(c0.owner).to eq(Fiber.current)
         expect(c1.owner).to eq(fiber)
-        expect(c0.in_use?).to be
-        expect(c1.in_use?).to be
       end
 
       it "should reclaim connections when the fiber has exited" do
-        expect(client).to receive(:query) { }.exactly(2).times
-
-        reap_connection_count = Rails::VERSION::MAJOR > 4 ? 5 : 3
-        expect(ActiveRecord::Base.connection_pool).to receive(:reap_connections).with(no_args).exactly(reap_connection_count).times.and_call_original
```

Comment:
This expect on the query isn't a meaningful test and changes across rails versions. The main test is below.